src/webextension/background/views.js

```diff
@@ -0,0 +1,98 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+
+const views = {
+  "firstrun": null,
+  "manage": [],
+};
+
+(function setupFirstRun(views) {
+  let tabId;
+
+  views.firstrun = {
+    open: async() => {
+      let windowId;
+      if (tabId !== undefined) {
+        // verify tab still exists
+        try {
+          let tabInfo = await browser.tabs.get(tabId);
+          windowId = tabInfo.windowId;
+        } catch (err) {
+          // does not exist, forget existing info
+          tabId = undefined;
+        }
+      }
+
+      if (tabId !== undefined) {
+        // focus owning window and activate tab
+        await browser.windows.update(windowId, {
+          focused: true,
+        });
+        await browser.tabs.update(tabId, {
+          active: true,
+        });
+      } else {
+        // create a tab in the current window
+        let tabInfo = await browser.tabs.create({
+          url: browser.extension.getURL("/firstrun/index.html"),
+        });
+        tabId = tabInfo.id;
+      }
+      return tabId;
+    },
+    close: async() => {
+      try {
+        tabId && await browser.tabs.remove(tabId);
+      } catch (err) {
+        // TODO: loggit?
+        console.log(`could not close firstrun view: ${err.message}`);
+      }
+      tabId = undefined;
+    },
+  };
+})(views);
+
+(function setupManage(views) {
+  let all = new Set();
+
+  views.manage = {
+    open: async() => {
+      // assume we always want to open a new one
+      let tabInfo = await browser.tabs.create({
+        url: browser.extension.getURL("/manage/index.html"),
+      });
+      all.add(tabInfo.id);
+      return tabInfo.id;
+    },
+    close: async() => {
+      let pending = [...all.values()].map(async(tabId) => {
+        try {
+          tabId && browser.tabs.remove(tabId);
```

Comment:
Ditto here; are we sure `tabId` can never be 0?

Reply:
[ see above ]